common/utils/state-gen.ts

```diff
@@ -110,20 +111,16 @@ function setupEcsForPlayer(
 		options.startWithAllCards || options.unlimitedCards ? cards.length : 7
 
 	if (options.shuffleDeck) {
-		cards
-			.sort(() => Math.random() - 0.5)
-			.forEach((card, i) => {
-				if (card.slot.inDeck()) card.slot.order = i
-			})
+		fisherYatesShuffle(cards).forEach((card, i) => {
+			if (card.slot.inDeck()) card.slot.order = i
+		})
 
 		while (
 			!cards.slice(0, amountOfStartingCards).some((card) => card.isHermit())
 		) {
-			cards
-				.sort(() => Math.random() - 0.5)
-				.forEach((card, i) => {
-					if (card.slot.inDeck()) card.slot.order = i
-				})
+			fisherYatesShuffle(cards).forEach((card, i) => {
+				if (card.slot.inDeck()) card.slot.order = i
+			})
```

Comment:
Can you rewrite this to use a `do...while` loop instead of repeating the same shuffle logic

Reply:
No, I do not like do whiles.